vcr/util.py

```diff
@@ -92,6 +92,21 @@ def composed(incoming):
 def read_body(request):
     if hasattr(request.body, "read"):
         return request.body.read()
+    if hasattr(request.body, "__iter__") and not isinstance(
+        request.body,
+        (bytearray, bytes, dict, list, str),
+    ):
+        body = list(request.body)
+        if body:
+            if isinstance(body[0], str):
+                return "".join(body).encode("utf-8")
+            elif isinstance(body[0], bytes):
+                return b"".join(body)
+            elif isinstance(body[0], int):
+                return bytes(body)
+            else:
```

Comment:
I wonder if we should append `# pragma: no cover` to this line or add a test instead. What do you think?

Reply:
added a test.